news/2022-01-24-weld-500Beta1.asciidoc

```diff
@@ -0,0 +1,121 @@
+Therefore, here we are with a proper release of Weld API `5.0.Beta5` and Weld Core `5.0.0.Beta1` which implements all the bits and pieces of CDI 4.0 plus brings a few more changes and deprecation removals.
+
+This post will try to cover Weld-side of things meaning anything that changed and is Weld specific or influences integrators in any way.
+If you want to learn what's new in CDI 4.0, I suggest taking a look at link:https://eclipse-ee4j.github.io/cdi/2021/10/25/way-to-cdi4.html[this detailed article] I wrote some time ago.
+It provides a good overview of all major changes, all of which are implemented in this Weld release.
+In case you want to check out the list of JIRA issues for Weld releases, the last chapter will provide links for both Alpha releases as well as this Beta release.
+Last but not least, you can also reach out to us via link:https://gitter.im/weld/user[Gitter channel] or link:https://lists.jboss.org/admin/lists/weld-dev.lists.jboss.org/[`weld-dev@lists.jboss.org` mailing list].
+
+Without further due, let's dive into it!
+
+=== Target JDK version for Weld 5 (and for CDI 4) is now 11
+
+Therefore, JDK 11 is now the minimal version required to run Weld.
+This is basically a requirement for all Jakarta EE 10 specifications.
+On the other hand, we now also test against and support JDK 17 as well.
+
+
+=== Weld 5 is no longer a multi-release (MR) JARs
+
+As you might know, Weld 3 and 4 are running some different class defining strategies for JDK 8 or 11+.
+Since JDK 11 is the basis, we no longer need to do that and were able to remove some hacks from code.
+
+=== Deprecations and removals
+
+Here is a list of the changes of this nature:
+
+* Multiple methods from `org.jboss.weld.serialization.spi.ProxyServices` were removed.
+** If you are an integrator, you probably implemented this interface at some point in time; these methods weren't used by Weld anyway, so they are now gone for good.
+** Here is a link:https://issues.redhat.com/browse/WELD-2702[link to the issue removing it].
+* Any integrators basing their implementations of Jandex likely implemented `ClassFileInfo`; its `isTopLevelClass()` has been removed.
+** Note that this method was redundant and can be replaced by `getClassNestingType()`.
+** This was removed as part of the above mentioned issue.
+* `WeldInstance.Handler` class is now deprecated and so are multiple methods returning this type from `WeldInstance`
+** The reasons for this removal is that the very same feature was accepted into CDI specification so you can now use CDI API interfaces to achieve the same.
+** For more information, take a look at link:https://github.com/eclipse-ee4j/cdi/blob/master/api/src/main/java/jakarta/enterprise/inject/Instance.java#L266-L315[`Instance.Handle` class from CDI API].
+** And again, a link:https://issues.redhat.com/browse/WELD-2685[link to the relevant issue].
+
+
+=== Inclusion of CDI Lite within Weld
```

Comment:
I'm still struggling to understand the nomenclature of CDI 4 ;-) but shouldn't we rather mention the build-time extensions API in the title?

Reply:
Ah, it's mentioned below... now I'm confused ;-)

Reply:
I just wanted to point out that CDI Lite is nothing special and is part of Weld by default.
Just in case users read about Lite from other posts and are unclear in terms of what is supported by Weld impl.

Reply:
I see. Maybe I'd choose another title then... something like "Weld implements CDI Lite"